models/vyos/ntp/vyos_ntp.yaml

```diff
@@ -0,0 +1,61 @@
+---
+
+GENERATOR_VERSION: '1.0'
+ANSIBLE_METADATA: |
+  {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'network'
+  }
+
+NETWORK_OS: vyos
+RESOURCE: ntp
+COPYRIGHT: Copyright 2021 Red Hat
+
+
+DOCUMENTAION: |
+  module: vyos_ntp
+  version added:
+  short_description:  Manages ntp modules of Vyos
+  description: This module manages ntp configuration on devices running Vyos
+  author: Varshitha Yataluru (@vyatalur)
+  options:
+    config:
+        description: List of configurations for ntp module
+        type: list
+        elements: dict
+        suboptions:
+            allow-clients:
+                description: Network Time Protocol (NTP) server options
+                type: str
+                suboptions:
```

Comment:
@YVarshitha type `str` is not supposed to have suboptions, as we discussed last day. You can have `choices`.
But in this case, as you have options under it should be `dict/list`